internal/index/indexscanner/scanner_test.go

```diff
@@ -90,6 +94,58 @@ func TestReadPluginFile(t *testing.T) {
 	}
 }
 
+func TestReadReceiptFromFile(t *testing.T) {
+	type args struct {
+		receiptFilePath string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantStatus index.ReceiptStatus
+	}{
+		{
+			name: "read receipt of plugin without status",
+			args: args{
+				receiptFilePath: filepath.Join(testdataPath(t), "testindex", "receipts", "foo.yaml"),
+			},
+			wantStatus: index.ReceiptStatus{},
+		},
+		{
+			name: "read receipt of plugin from default index with status",
+			args: args{
+				receiptFilePath: filepath.Join(testdataPath(t), "testindex", "receipts", "foo-explicit.yaml"),
+			},
+			wantStatus: index.ReceiptStatus{
+				Source: index.SourceIndex{
+					Name: constants.DefaultIndexName,
+				},
+			},
+		},
```

Comment:
I think this and the next test case are pretty much identical. I wouldn't use the `default` index as an explicitly given status value, because it is somewhat special. Just use something that is unlikely to be used by accident, like `test-custom` and everything is fine.

Then you can also get rid of one of the two additional receipt yamls.

Reply:
I wanted to test 3 types of receipts in this: one from a custom index, one from the default index before migration, and one from the default index after migration (with/without status object). If that seems excessive I can remove some of them.

Reply:
Honestly, `name: custom` and `name: default` are equivalent things from what they're testing. So one of them is redundant.

Similarly, not having `status:` field is _expected_ to yield the zero-value of that field (that's just what YAML/JSON libraries do). So we aren't testing anything valuable there either.

All I'm interested in testing is "round-tripping the YAML parsing of field names", which can be done in 1 file with:

```
[...]
status:
  source:
    name: foo
```

Reply:
Sounds good, I'll simplify the tests.